src/gui/properties/speedplotview.h

```diff
@@ -86,6 +90,36 @@ class SpeedPlotView : public QGraphicsView
     void paintEvent(QPaintEvent *event) override;
 
 private:
+    class Averager
+    {
+    public:
+        Averager(int _divider, boost::circular_buffer<PointData> &_sink)
+            : cnt(0)
+            , divider(_divider)
+            , sink(_sink)
+        {acc.reset();};
+        void push(const PointData &p)
+        {
+            acc.x += p.x;
+            for (int id = UP; id < NB_GRAPHS; ++id)
+                acc.y[id] += p.y[id];
+            cnt = (cnt + 1) % divider;
+            if (cnt==0) {
+                for (int id = UP; id < NB_GRAPHS; ++id)
+                    acc.y[id] /= divider;
+                acc.x /= divider;
+                // now flush out averaged data
+                sink.push_back(acc);
+                acc.reset();
+            }
+        };
+        int cnt;
+
+    private:
+        int divider;
+        boost::circular_buffer<PointData> &sink;
+        PointData acc;
```

Comment:
As I see this is only used in push() where it is copied into a circular_buffer and then reset.
Why don't you just create it as local variable at the start of the push() function? It will also eliminate the hacky implementation of the reset() function.

Reply:
Oops. Sorry, I misread it.
Once you address the other comments, I'll see it again.